Upstream patch from
https://gitlab.gnome.org/GNOME/gnome-keyring/-/merge_requests/62
which was merged for release 46.1

From f298c87e14e89656992ccfd17b133f4f2ab9c5ea Mon Sep 17 00:00:00 2001
From: Yaakov Selkowitz <yselkowi@redhat.com>
Date: Wed, 24 Jan 2024 14:01:06 -0500
Subject: [PATCH] gkm: fix test-sexp with GCC 14
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

pkcs11/gkm/test-sexp.c: In function ‘test_sign_verify’:
pkcs11/gkm/test-sexp.c:231:92: error: passing argument 6 of ‘gkm_crypto_sign_xsa’ from incompatible pointer type [-Wincompatible-pointer-types]
In file included from pkcs11/gkm/test-sexp.c:31:
./pkcs11/gkm/gkm-crypto.h:92:86: note: expected ‘CK_ULONG_PTR’ {aka ‘long unsigned int *’} but argument is of type ‘gsize *’ {aka ‘unsigned int *’}
pkcs11/gkm/test-sexp.c:251:87: error: passing argument 6 of ‘gkm_crypto_sign_xsa’ from incompatible pointer type [-Wincompatible-pointer-types]
./pkcs11/gkm/gkm-crypto.h:92:86: note: expected ‘CK_ULONG_PTR’ {aka ‘long unsigned int *’} but argument is of type ‘gsize *’ {aka ‘unsigned int *’}
pkcs11/gkm/test-sexp.c:271:91: error: passing argument 6 of ‘gkm_crypto_sign_xsa’ from incompatible pointer type [-Wincompatible-pointer-types]
./pkcs11/gkm/gkm-crypto.h:92:86: note: expected ‘CK_ULONG_PTR’ {aka ‘long unsigned int *’} but argument is of type ‘gsize *’ {aka ‘unsigned int *’}

Signed-off-by: Yaakov Selkowitz <yselkowi@redhat.com>
---
 pkcs11/gkm/test-sexp.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/pkcs11/gkm/test-sexp.c b/pkcs11/gkm/test-sexp.c
index ba104a0e5..9b663cc40 100644
--- a/pkcs11/gkm/test-sexp.c
+++ b/pkcs11/gkm/test-sexp.c
@@ -221,10 +221,10 @@ test_sign_verify (Test *test, gconstpointer unused)
 {
 	gcry_sexp_t pubkey = NULL;
 	gboolean ret;
-	guchar data[] = TEST_DATA;
-	guchar data_size = TEST_DATA_SIZE;
-	guchar signature[128];
-	gsize signature_size = 128;
+	CK_BYTE data[] = TEST_DATA;
+	CK_ULONG data_size = TEST_DATA_SIZE;
+	CK_BYTE signature[128];
+	CK_ULONG signature_size = 128;
 
 	/* RSA */
 	/* sign some data */
-- 
GitLab

